parts/openshift/unstable/openshiftmasterscript.sh

```diff
@@ -32,10 +32,33 @@ else
 	COCKPIT_VERSION="latest"
 fi
 
-# TODO: with WALinuxAgent>=v2.2.21 (https://github.com/Azure/WALinuxAgent/pull/1005)
-# we should be able to append context=system_u:object_r:container_var_lib_t:s0
-# to ResourceDisk.MountOptions in /etc/waagent.conf and remove this stanza.
+if grep -q ^ResourceDisk.Filesystem=xfs /etc/waagent.conf; then
+	# Bad image: docker and waagent are racing.  Try to fix up.  Leave this code
+	# until the bad images have gone away.
+	set +e
+
+	# stop docker if it hasn't failed already
+	systemctl stop docker.service
+
+	# wait until waagent has run mkfs and mounted /var/lib/docker
+	while [[ $(stat -c %D /var/lib/docker) != 811 ]]; do
```

Comment:
where does 811 come from it doesn't seem static? When I boot a 3.10 cluster I'm seeing:
```
[root@ocp-master-25550318-0 ~]# stat -c %D /var/lib/docker
802
```
Is `mountpoint /var/lib/docker/containers` a good check for knowing it is mounted and `mount -l -t xfs` a better check to know we lost?  

```
[root@ocp-master-25550318-0 ~]# mountpoint /var/lib/docker/containers
/var/lib/docker/containers is a mountpoint
[root@ocp-master-25550318-0 ~]# mount -l -t xfs
/dev/sda1 on / type xfs (rw,relatime,seclabel,attr2,inode64,noquota)
/dev/sda2 on /var type xfs (rw,relatime,seclabel,attr2,inode64,grpquota)
/dev/sda2 on /var/lib/docker/containers type xfs (rw,relatime,seclabel,attr2,inode64,grpquota)
/dev/sda2 on /var/lib/docker/overlay2 type xfs (rw,relatime,seclabel,attr2,inode64,grpquota)
```

Reply:
actually, maybe it's my input file on the device number but I guess the question is still valid.

Reply:
`mountpoint /var/lib/docker/containers` also works fine, thanks - it's problematic when you know your system calls better than util-linux :/
updated.
I don't see where `mount -l -t xfs` would be needed / what it replaces?

Reply:
> I don't see where mount -l -t xfs would be needed / what it replaces?

was curious if we could tell from that call output if the race was lost or not.  Sounds like not.